src/main/java/seedu/address/logic/commands/MarkSupplierCommand.java

```diff
@@ -43,7 +43,11 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(Messages.MESSAGE_INVALID_SUPPLIER_DISPLAYED_INDEX);
         }
 
-        Supplier supplierToMark = model.getModifiedSupplierList().get(targetIndex.getZeroBased());
+        Supplier supplierToMark = model.getFilteredSupplierList().get(targetIndex.getZeroBased());
```

Comment:
should we be retrieving the supplier from the modified list instead of the filtered list?

Reply:
Fixed!